File: 2022/Python/day2/part1.py
from os import path
basepath = path.dirname(__file__)

def playGame(opp, me):
    oppIdx = ord(opp) - ord('A')
    meIdx = ord(me) - ord('X')

    points = meIdx + 1
    if oppIdx == meIdx:
        points += 3
    elif (oppIdx + 1) % 3 == meIdx:
        points += 6
    return points

score = 0
with open(path.join(basepath, "input.txt")) as f:
    lines = f.readlines()
    for line in lines:
        opp, me = line.strip().split(" ")
        score += playGame(opp, me)

print(f"total score: {score}")

File: 2022/JavaScript/day2/part1.js
const text = await Deno.readTextFile("./input.txt");

const moves = {
  X: { score: 1, beats: "C", tiesWith: "A" },
  Y: { score: 2, beats: "A", tiesWith: "B" },
  Z: { score: 3, beats: "B", tiesWith: "C" },
};

let total = 0;

const lines = text.split("\n");
for (const line of lines) {
  const [opp, you] = line.split(" ");
  total += moves[you].score;
  if (opp === moves[you].beats) {
    total += 6;
  } else if (opp === moves[you].tiesWith) {
    total += 3;
  }
}

console.log(total);
